File: yh_tutorial/yh_tutorial_2/src/yh_sub_2_int.cpp
#include "ros/ros.h"
#include "yh_tutorial_2/yh_msg_2.h"

void msgCallback(const yh_tutorial_2::yh_msg_2::ConstPtr &msg)
{
    int n = msg->data;
    if (n%2 == 1)
    {
        ROS_INFO("%d", n);
    }
}

int main(int argc, char **argv)
{
    ros::init(argc, argv, "yh_sub_2_int");
    ros::NodeHandle nh;
    ros::Subscriber sub = nh.subscribe("yh_topic_2", 100, msgCallback);

    ros::spin();

    return 0;
}

File: yh_tutorial/yh_tutorial_2/scripts/yh_sub_2_int.py
#!/usr/bin/python
# -*- coding: utf8 -*-

import rospy
from yh_tutorial_2.msg import yh_msg_2

def msgCallback(msg):
    if msg.data % 2 == 1:
        rospy.loginfo("%d", msg.data)

def listener():
    rospy.init_node("yh_sub_2_int", anonymous=True)
    rospy.Subscriber("yh_topic_2", yh_msg_2, msgCallback)
    rospy.spin()

if __name__ == "__main__":
    listener()